src/main/java/casper/CasperBot.java

```diff
@@ -0,0 +1,199 @@
+                            if (deadline == null || deadline.isEmpty()) {
+                                throw new CasperBotMissingInputException("/by", "Deadline");
+                            }
+                            LocalDate dateOfDeadline = LocalDate.parse(deadline);
+                            Deadline newDeadline = new Deadline(deadlineDescription, false, dateOfDeadline);
+                            this.taskList.addTask(newDeadline);
+                            this.storage.writeToFile(newDeadline);
+                            this.ui.addTaskMessage(newDeadline);
+                            break;
+                        case EVENT:
+                            String eventDescription = hashMap.get("description");
+                            if (eventDescription.isEmpty()) {
+                                throw new CasperBotMissingInputException("description", "Event");
+                            }
+                            String start = hashMap.get("from");
+                            if (start == null || start.isEmpty()) {
+                                throw new CasperBotMissingInputException("/from", "Event");
+                            }
+                            LocalDate dateOfStart = LocalDate.parse(start);
+                            String end = hashMap.get("to");
+                            if (end == null || end.isEmpty()) {
+                                throw new CasperBotMissingInputException("/to", "Event");
+                            }
+                            LocalDate dateOfEnd = LocalDate.parse(end);
+                            Event newEvent = new Event(eventDescription, false, dateOfStart, dateOfEnd);
+                            this.taskList.addTask(newEvent);
+                            this.storage.writeToFile(newEvent);
+                            this.ui.addTaskMessage(newEvent);
+                            break;
+                        }
+                        this.ui.printTaskListLength(this.taskList.getNumberOfTasks());
+                    } catch (DateTimeParseException e) {
+                        throw new CasperBotInvalidDateException();
+                    }
+                }
+                else if (inputArray[0].equalsIgnoreCase("bye")) {
+                    this.ui.exit();
+                } else {
+                    throw new CasperBotInvalidCommandException();
+                }
+            } catch (CasperBotException e) {
+                this.ui.showErrorMessage(e);
+            } finally {
+                this.ui.printLine();
+            }
+        }
+        scanner.close();
+    }
+
+    /**
+     * Checks if a string is a valid command
+     *
+     * @param command The string to be validated
+     * @param commandType The type of command it should belong to (CREATE, TASK)
+     * @return
+     */
+    private static boolean isValidCommand(String command, CommandType commandType) {
```

Comment:
I think method names should be in the form of a verb, `isValidCommand` is more suitable to be a boolean variable name.